cmd/kube-proxy/app/server_test.go

```diff
@@ -279,35 +281,130 @@ nodePortAddresses:
 
 // TestLoadConfigFailures tests failure modes for loadConfig()
 func TestLoadConfigFailures(t *testing.T) {
+	duplicateKeyYAMLTemplate := `bindAddress: 0.0.0.0
+bindAddress: 1.2.3.4
+clientConnection:
+  acceptContentTypes: "abc"
+  burst: 100
+  contentType: content-type
+  kubeconfig: "/path/to/kubeconfig"
+  qps: 7
+clusterCIDR: "1.2.3.0/24"
+configSyncPeriod: 15s
+conntrack:
+  maxPerCore: 2
+  min: 1
+  tcpCloseWaitTimeout: 10s
+  tcpEstablishedTimeout: 20s
+healthzBindAddress: "1.2.3.4:12345"
+hostnameOverride: "foo"
+iptables:
+  masqueradeAll: true
+  masqueradeBit: 17
+  minSyncPeriod: 10s
+  syncPeriod: 60s
+ipvs:
+  minSyncPeriod: 10s
+  syncPeriod: 60s
+  excludeCIDRs:
+    - "10.20.30.40/16"
+    - "fd00:1::0/64"
+kind: KubeProxyConfiguration
+metricsBindAddress: "1.2.3.4:12346"
+mode: "loopback"
+oomScoreAdj: 17
+portRange: "2-7"
+udpIdleTimeout: 123ms
+nodePortAddresses:
+  - "10.20.30.40/16"
+  - "fd00:1::0/64"`
+
+	unknownFieldYAMLTemplate := `bindAddress: 0.0.0.0
+foo: bar
+clientConnection:
+  acceptContentTypes: "abc"
+  burst: 100
+  contentType: content-type
+  kubeconfig: "/path/to/kubeconfig"
+  qps: 7
+clusterCIDR: "1.2.3.0/24"
+configSyncPeriod: 15s
+conntrack:
+  maxPerCore: 2
+  min: 1
+  tcpCloseWaitTimeout: 10s
+  tcpEstablishedTimeout: 20s
+healthzBindAddress: "1.2.3.4:12345"
+hostnameOverride: "foo"
+iptables:
+  masqueradeAll: true
+  masqueradeBit: 17
+  minSyncPeriod: 10s
+  syncPeriod: 60s
+ipvs:
+  minSyncPeriod: 10s
+  syncPeriod: 60s
+  excludeCIDRs:
+    - "10.20.30.40/16"
+    - "fd00:1::0/64"
+kind: KubeProxyConfiguration
+metricsBindAddress: "1.2.3.4:12346"
+mode: "loopback"
+oomScoreAdj: 17
+portRange: "2-7"
+udpIdleTimeout: 123ms
+nodePortAddresses:
+  - "10.20.30.40/16"
+  - "fd00:1::0/64"`
+
 	testCases := []struct {
 		name   string
 		config string
-		expErr string
+		errFn  func(err error) bool
 	}{
 		{
 			name:   "Decode error test",
 			config: "Twas bryllyg, and ye slythy toves",
-			expErr: "could not find expected ':'",
+			errFn:  func(err error) bool { return err != nil },
 		},
 		{
 			name:   "Bad config type test",
 			config: "kind: KubeSchedulerConfiguration",
-			expErr: "no kind",
+			errFn:  func(err error) bool { return err != nil },
 		},
 		{
 			name:   "Missing quotes around :: bindAddress",
 			config: "bindAddress: ::",
-			expErr: "mapping values are not allowed in this context",
+			errFn:  func(err error) bool { return err != nil },
+		},
+		{
+			name:   "Duplicate fields",
+			config: duplicateKeyYAMLTemplate,
+			errFn: func(err error) bool {
```

Comment:
This can change to:
```go
errFn: kuberuntime.IsStrictDecodingError
```
That function contains a nil check, and it should already have been asserted an error exists before running it  because of the nature of this range test.

Reply:
Changed

Reply:
Fixed